### PR TITLE
avoid oscilation between ORAT and LRAT when they are equal

### PR DESCRIPTION
**Pull request by totto82**
Preparation for getting #4094 in.

### CODE REVIEW

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
I have analyzed the test failures. There are differences due to differences in time-stepping that are again due to differences in number of newton iterations. This PR stabilize cases where ORAT = LRAT and water rate is epsilon. The output is a bit noisy so  I will move it to the debug.  

image

**Comment by totto82** [comment]
jenkins build this please

File: opm/simulators/wells/BlackoilWellModelGeneric.cpp [hunk 1]
```diff
@@ -996,7 +996,17 @@ checkGroupProductionConstraints(const Group& group,
             // sum over all nodes
             current_rate = comm_.sum(current_rate);
 
-            if (controls.liquid_target < current_rate  ) {
+            bool skip = false;
+            if (controls.liquid_target == controls.oil_target) {
+                double current_water_rate = WellGroupHelpers::sumWellSurfaceRates(group, schedule(), well_state, reportStepIdx, phase_usage_.phase_pos[BlackoilPhases::Aqua], false);
+                current_water_rate = comm_.sum(current_water_rate);
```
Comment (atgeirr) [inline opm/simulators/wells/BlackoilWellModelGeneric.cpp:1002]: Have we not communicated and stored the group rates in the well state, is it necessary to do that again here?

**Review by atgeirr — COMMENTED** [review]
The logic looks good to me, I only have one question for the group case.

**Review by atgeirr — APPROVED** [review]

**Comment by totto82** [comment]
jenkins build this update_data please

**Comment by totto82** [comment]
jenkins build this opm-tests=810 please

**Comment by bska** [comment]
The updated reference solutions have been installed on the CI system.  I'm merging this now per prior approval.

**Pull request merged**